creator/ingest_runs/genomic_data_loader.py

```diff
@@ -100,6 +105,61 @@ def load_harmonized_genomic_files(self, manifest_df):
 
         return genomic_df
 
+    def _check_and_merge_s3_scrape(self, manifest_df, file_df):
```

Comment:
The business logic in each of these check and merge functions looks good. 

There is quite a bit of repetition though in each check and merge function so it would be really nice if you could dry it up with one function. It'll be way easier to test that as well bc you just need a test for the single merge function, and then in all of your load function tests you could simply check that the merge function was called with the right args. 

Something like this could work: 

```python
def merge_and_validate(
    ingest_run, left_df, right_df, left_df_name, right_df_name, custom_message=None, **outer_merge_kwargs
):
    """
    Merge two DataFrames together and report any missing data. Missing
    data is defined as any records that are in the left DataFrame but 
    not in the right DataFrame.
    """
    outer_merge_kwargs["with_merge_detail_dfs"] = True
    inner_merge_df, left_only, _ = outer_merge(
        left_df,
        right_df,
        **outer_merge_kwargs
    )
    key = None
    for kwarg in ["on", "left_on", "right_on"]:
        key = outer_merge_kwargs.get(kwarg)
        if key:
            break
    missing = left_only[key].drop_duplicates()
    if not missing.empty:
        num_missing = missing.shape[0]
        message = (
            f"During the ingest GWO run: {ingest_run.pk}, "
            f"{num_missing} records in the {left_df_name} "
            f"were not found in the {right_df_name}:"
            f"\n{pformat(missing_files)}"
        )
        create_missing_event(message, ingest_run)
        logger.warning(message)
        logger.warning(custom_message)

    return inner_merge_df

```

This also eliminates the problem of having to check if `ingest_run` exists or not. In your unittest, you can simply pass in a mock of ingest_run which just needs to have the `pk` and `creator` attributes since those are what you use in log messages and in the `create_event` method.